src/content/docs/en/reference/container-reference.mdx

```diff
@@ -165,6 +165,66 @@ const result = await container.renderToString(Card, {
 });
 ```
 
+### `props` option
+
+**Type**: `Record<string, unknown>`
+
+An option to pass [`properties`](en/basics/astro-components/#props) for astro components.
+
+```js name="Card.test.js"
+import Card from "../src/components/Card.astro";
+
+const result = await container.renderToString(Card, { 
+  props: { name: "Hello, world!" },
+});
+```
+
+```astro name="Card.astro"
+---
+// For TypeScript support
+interface Props {
+  name: string;
+};
+
+const { name } = Astro.props;
+---
+<div>
+  {name}
+</div>
+```
+
+> Note: The type-hints are not automatically recognised, so you would need to share the
+> interface outside of `.astro` files in order to be available for type-hints:
+
+```js name="Card.test.js"
+import Card from "../src/components/Card.astro";
+import { CardProps } from "../src/components/types.ts";
+
+const props: CardProps = { name: "Hello, world!" };
+const result = await container.renderToString(Card, { 
+  props,
+});
+```
+
+```astro name="types.ts"
+export interface CardProps {
+  name: string;
+};
+```
+
+```astro name="Card.astro"
+---
+import { CardProps } from "./types.ts";
+
+interface Props extends CardProps {};
+
+const { name } = Astro.props;
+---
+<div>
+  {name}
+</div>
+```
```

Comment:
``````suggestion
#### Type-hints
 
When using the Container API outside of `.astro` files (e.g. `test.js`), define and import any interfaces to make them available for type-hints:

```js name="Card.test.js"
import Card from "../src/components/Card.astro";
import { CardProps } from "../src/components/types.ts";

const props: CardProps = { name: "Hello, world!" };
const result = await container.renderToString(Card, { 
  props,
});
```

```astro name="types.ts"
export interface CardProps {
  name: string;
};
```

```astro name="Card.astro"
---
import { CardProps } from "./types.ts";

interface Props extends CardProps {};

const { name } = Astro.props;
---
<div>
  {name}
</div>
```
``````
I'll need your help to make sure the wording is accurate, but we try not to use "notes" etc. for things that are just regular parts of the documentation. Notes, cautions, tips are really for "out of the ordinary" content.

So, I'm assuming it makes sense that instead of a note, we can have a section heading. The advantage here is that these get indexed, are easy to link to for support etc. vs being in a note.

With that being said, how do you feel about the suggestion above? What should be changed to make it completely accurate/helpful?

Reply:
Hmm, the only thing I see now is not doing
```typescript
import { type CardProps } from "./types.ts";
```

Reply:
Incorporated the suggestion and added a little more at the end to encourage astro component maintainers to expose their properties through interfaces.